week2/materials/homework/webapp/src/components/Products.js

```diff
@@ -1,15 +1,29 @@
 import useProducts from "../hooks/useProducts";
+import useNotifications from "../hooks/useNotification";
 
 function Products() {
-  const { products, cart, addProduct, removeProduct } = useProducts();
+  const { products, cart, addProduct, Total, removeProduct } = useProducts();
+  const { notifications, createNotification } = useNotifications();
 
   const isInCart = (product) => {
     return !cart.find((item) => item.id === product.id);
   };
 
+  const productAddInCart =  (product)=>{
+    addProduct(product);
+    // console.log('product added');
+    createNotification(`${product.name} added to your cart`);
+  }
+
+  const productRemoveFrmCart = (product)=>{
+    removeProduct(product);
+    createNotification(`${product.name} removed from your cart`)
+  }
+  
   return (
     <div>
       <div className="row">
+      {notifications && <h2>Notification: {notifications}</h2>}
```

Comment:
Maybe `notifications` should be `notification` as it is singular and just a string, I almost got confused that it might be an array and then this would not work